File: js/code_challenges/OOPiracy.js
// https://www.codewars.com/kata/54fe05c4762e2e3047000add/javascript

function Ship(draft, crew) {
  this.draft = draft;
  this.crew = crew;

  this.isWorthIt = function () {
    const weight = this.draft - this.crew * 1.5;
    console.log(weight)
    return weight >= 20;
  };
}

const canoe = new Ship(10, 2)
console.log(canoe.isWorthIt())

const titanic = new Ship(30, 6)
console.log(titanic.isWorthIt())

const cruise = new Ship(25, 4)
console.log(cruise.isWorthIt())

File: ruby/coding challenges/oopiracy.rb
# https://www.codewars.com/kata/54fe05c4762e2e3047000add/train/ruby

class Ship
def initialize(draft,crew)
    @draft=draft
    @crew=crew
  end
  def is_worth_it
    weight = @draft - (@crew * 1.5)
    if weight > 20
      return true
    else
      return false
    end
  end
end